saw-core/src/SAWCore/Name.hs

```diff
@@ -285,6 +296,62 @@ scFreshNameURI nm i = fromMaybe (panic "scFreshNameURI" ["Failed to construct na
        }
 
 
+-- Display Name Environments --------------------------------------------------------
+
+-- | A 'DisplayNameEnv' encodes the mappings between display names (type
+-- 'Text') and internal names (type 'VarIndex'). Multiple display
+-- names may be associated with each internal name; these are stored
+-- in priority order, with preferred display names first.
+data DisplayNameEnv =
+  DisplayNameEnv
+  { displayNames :: !(IntMap [Text]) -- Keyed by VarIndex; preferred names come first.
```

Comment:
Is the list of `[Text]` here always non-empty? If so, it would be worth stating that as an invariant (or even converting the type to `NonEmpty Text`).